nixos/modules/services/web-apps/ocis.nix

```diff
@@ -0,0 +1,114 @@
+{
+  config,
+  lib,
+  pkgs,
+}:
+
+let
+  inherit (lib) types;
+  cfg = config.services.ocis;
+in
+
+{
+  options = {
+    services.ocis = {
+      enable = lib.mkEnableOption "ownCloud Infinite Scale";
+
+      package = lib.mkPackageOption pkgs "ocis-bin" { };
+
+      environmentFile = lib.mkOption {
+        type = types.nullOr types.path;
+        default = null;
+        example = "/run/keys/ocis.env";
+        description = ''
+          An environment file as defined in {manpage}`systemd.exec(5)`.
+        '';
+      };
+
+      settings = lib.mkOption {
+        type = lib.types.submodule {
+          freeformType =
+            with lib.types;
+            attrsOf (
+              let
+                typeList = [
+                  bool
+                  float
+                  int
+                  str
+                  path
+                  package
+                ];
+              in
+              oneOf (
+                typeList
+                ++ [
+                  (listOf (oneOf typeList))
+                  (attrsOf (oneOf typeList))
+                ]
+              )
+            );
+
+          options = {
+            address = lib.mkOption {
+              type = types.str;
+              default = "127.0.0.1";
+              description = "Web interface address.";
+            };
+
+            port = lib.mkOption {
+              type = types.port;
+              default = 9200;
+              description = "Web interface port.";
+            };
+
+            stateDir = lib.mkOption {
+              default = "/var/lib/ocis";
+              type = types.str;
+              description = "Workout-tracker data directory.";
+            };
+          };
+        };
+        default = { };
+        description = ''
+          Extra config options.
+        '';
+        example = {
+          OCIS_URL = "https://some-hostname-or-ip:9200";
+          PROXY_HTTP_ADDR = "0.0.0.0:9200";
+          OCIS_INSECURE = "false";
+          OCIS_LOG_LEVEL = "error";
+          OCIS_CONFIG_DIR = "/etc/ocis";
+          OCIS_BASE_DATA_PATH = "/var/lib/ocis";
+        };
+      };
+    };
+  };
+
+  config = lib.mkIf cfg.enable {
+    systemd.services.ocis = {
+      description = "ownCloud Infinite Scale Stack";
+      wantedBy = [ "multi-user.target" ];
+      environment = {
+        PROXY_HTTP_ADDR = "${cfg.settings.address}:${toString cfg.settings.port}";
+        OCIS_INSECURE = "false";
+        OCIS_LOG_LEVEL = "error";
```

Comment:
These should be overrideable by the user's settings.

Reply:
Done